packages/common/src/operations.graphql

```diff
@@ -93,6 +93,58 @@ query invoice($id: String!) {
             numberOfPacks
             packSize
             note
+            location {
```

Comment:
Going to type out my problem here .. I *think* this should be a non-issue if and when(?) the talk about changing the API to not return errors on nested fields happens.. but in the meantime

The annoyance here is:

- I want to parse the query result with the generated type "InvoiceQuery" into an `Invoice`. An `Invoice` has fields like `lines: InvoiceLine[]` rather than `lines: InvoiceLineRespinse` (which is `ConnectorError | InvoiceLineConnector`) or something.. so we have to parse the lines as we want `location: Location` rather than `location: Location | NodeError` etc
- So, we start by parsing and asserting that there is an `InvoiceNode`
- Then we parse the lines. Which involves mapping and asserting the stock line. Then we do the location.
- The location, also has a field .. `stock`. Which is a `StockLineResponse` ..
- So, if we don't query for `stock`, then the response is technically not a `LocationResponse` and so generated types don't work well. And more than that.. if we don't query for the full `StockLine`, then the types still don't work

Hm.

I think this can be fixed with using fragments. If we used a fragment of the location without the `stock` field, we should be able to generate a type ... `LocationRowFragment` or something like that and use that type rather than `LocationResponse`.

Though I'm not sure if I should go down that route and instead just wait for the API change, or start writing my own types..

Reply:
Yeah.. think it'll likely be a while